pyowc/data/fetchers.py

```diff
@@ -19,6 +19,46 @@
 from ..configuration import print_cfg
 
 
+@lru_cache(maxsize=50)
+def read_mat_ctd(filepath):
+    """Cache the result of reading a ctd .mat file from disk."""
+    return read_mat(filepath)
+
+
+@lru_cache(maxsize=50)
+def read_mat_bottle(filepath):
+    """Load and transform a bottle .mat file."""
+    data = read_mat(filepath)
+
+    # if data dimensions don't match, transpose to avoid indexing issues
+    if (data['lat'].size == data['pres'].size and data['lat'].shape[1] == data['pres'].shape[1]):
+        data['pres'] = data['pres'].T
+        data['ptmp'] = data['ptmp'].T
+        data['sal'] = data['sal'].T
+        data['temp'] = data['temp'].T
+
+    return data
+
+
+@lru_cache(maxsize=50)
+def read_mat_argo(filepath, pa_float_name):
+    """Load and transform an Argo .mat file."""
+    data = read_mat(filepath)
+    # remove the argo float being analysed from the data
+    not_use = []
+    for i in range(0, data['lat'].__len__()):
```

Comment:
Why the use of `__len__()` here? Could we use `len(data['lat'])` or `data['lat'].size`?

Reply:
The code itself was simply moved into this function for caching. I avoided making any other changes to minimise the chance of introducing other regressions. I agree that this usage is non-standard and we should remove it, I'm just not sure if we want to do it as part of this PR.